### PR TITLE
pt-BR docs translations

### PR DESCRIPTION
**Pull request by rodrigobastosv**
## Status
**READY*

## Breaking Changes
NO

## Description
Translation of the docs to portuguese (pt-BR)

### CODE REVIEW

File: docs/pt-br/_sidebar.md [hunk 1]
```diff
@@ -7,6 +7,8 @@
     - [flutter_bloc](pt-br/flutterbloccoreconcepts.md)
   - [Arquitetura](pt-br/architecture.md)
   - [Testando](pt-br/testing.md)
+  - [Naming Conventions](blocnamingconventions.md)
+  - [FAQs](faqs.md)
```
Comment (felangel) [inline docs/pt-br/_sidebar.md]:
```suggestion
  - [FAQs](pt-br/faqs.md)
```

File: docs/pt-br/_sidebar.md [hunk 2]
```diff
@@ -7,6 +7,8 @@
     - [flutter_bloc](pt-br/flutterbloccoreconcepts.md)
   - [Arquitetura](pt-br/architecture.md)
   - [Testando](pt-br/testing.md)
+  - [Naming Conventions](blocnamingconventions.md)
```
Comment (felangel) [inline docs/pt-br/_sidebar.md]:
```suggestion
  - [Naming Conventions](pt-br/blocnamingconventions.md)
```

File: docs/pt-br/faqs.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+❔ **Pergunta**: Qual é a diferença entre Bloc e Redux?
+
+💡 **Resposta**:
+
+BLoC é um padrão de design definido pelas seguintes regras:
+
+1. A entrada e saída do BLoC são fluxos e sumidouros simples.
+2. As dependências devem ser injetáveis ​​e a Plataforma agnóstica.
+3. Nenhuma ramificação de plataforma é permitida
```
Comment (zepfietje) [inline docs/pt-br/faqs.md]:
```suggestion
3. Nenhuma ramificação de plataforma é permitida.
```

File: docs/pt-br/faqs.md [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+4. A implementação pode ser o que você quiser, desde que siga as regras acima.
+
+As diretrizes da interface do usuário são:
+
+1. Cada componente "suficientemente complexo" possui um BLoC correspondente
```
Comment (zepfietje) [inline docs/pt-br/faqs.md]:
```suggestion
1. Cada componente "suficientemente complexo" possui um BLoC correspondente.
```

File: docs/pt-br/faqs.md [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+2. Os componentes devem enviar entradas "como estão"
```
Comment (zepfietje) [inline docs/pt-br/faqs.md]:
```suggestion
2. Os componentes devem enviar entradas "como estão".
```

File: docs/pt-br/faqs.md [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+3. Os componentes devem mostrar as saídas o mais próximo possível de "como estão"
```
Comment (zepfietje) [inline docs/pt-br/faqs.md]:
```suggestion
3. Os componentes devem mostrar as saídas o mais próximo possível de "como estão".
```

File: docs/pt-br/faqs.md [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+4. Todas as ramificações devem ser baseadas em saídas booleanas simples de BLoC
```
Comment (zepfietje) [inline docs/pt-br/faqs.md]:
```suggestion
4. Todas as ramificações devem ser baseadas em saídas booleanas simples de BLoC.
```

**Comment by rodrigobastosv** [comment]
@felangel is there anything missing here?

**Comment by felangel** [comment]
Will take a look tomorrow and merge 👍 thanks so much for your hard work!

File: docs/pt-br/flutterbloccoreconcepts.md [hunk 8]
```diff
@@ -0,0 +1,365 @@
...
+  },
+  builder: (context, state) {
+    // return widget here based on BlocA's state
+  }
+)
+```
+
+### BlocProvider
+
+**BlocProvider** é um widget Flutter que fornece um Bloc para seus filhos via `BlocProvider.of <T> (context)`. Ele é usado como um widget de injeção de dependência (DI) para que uma única instância de um Bloc possa ser fornecida a vários widgets em uma subárvore.
```
Comment (zepfietje) [inline docs/pt-br/flutterbloccoreconcepts.md]:
```suggestion
**BlocProvider** é um widget Flutter que fornece um Bloc para seus filhos via `BlocProvider.of<T>(context)`. Ele é usado como um widget de injeção de dependência (DI) para que uma única instância de um Bloc possa ser fornecida a vários widgets em uma subárvore.
```

File: docs/pt-br/flutterbloccoreconcepts.md [hunk 9]
```diff
@@ -0,0 +1,365 @@
...
+  },
+  builder: (context, state) {
+    // return widget here based on BlocA's state
+  }
+)
+```
+
+### RepositoryProvider
+
+**RepositoryProvider** é um widget Flutter que fornece um repositório para seus filhos via `RepositoryProvider.of <T> (context)`. Ele é usado como um widget de injeção de dependência (DI) para que uma única instância de um repositório possa ser fornecida a vários widgets em uma subárvore. O `BlocProvider` deve ser usado para fornecer blocs, enquanto o` RepositoryProvider` deve ser usado apenas para repositórios.
```
Comment (zepfietje) [inline docs/pt-br/flutterbloccoreconcepts.md]:
```suggestion
**RepositoryProvider** é um widget Flutter que fornece um repositório para seus filhos via `RepositoryProvider.of<T>(context)`. Ele é usado como um widget de injeção de dependência (DI) para que uma única instância de um repositório possa ser fornecida a vários widgets em uma subárvore. O `BlocProvider` deve ser usado para fornecer blocs, enquanto o` RepositoryProvider` deve ser usado apenas para repositórios.
```

File: docs/pt-br/fluttercountertutorial.md [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+    );
+  }
+}
+```
+
+?> **Nota**: Podemos acessar a instância do `CounterBloc` usando o `BlocProvider.of <CounterBloc> (context)` porque envolvemos nossa `CounterPage` em um `BlocProvider`.
+
+?> **Nota**: Estamos usando o widget `BlocBuilder` do `flutter_bloc` para reconstruir nossa interface do usuário em resposta a alterações de estado (alterações no valor do contador).
+
+?> **Nota**: `BlocBuilder` usa um parâmetro opcional `bloc`, mas podemos especificar o tipo do blocc e o tipo do estado, e o `BlocBuilder` encontrará o bloc automaticamente, assim não precisamos explicitar use `BlocProvider.of <CounterBloc> (context)`.
```
Comment (zepfietje) [inline docs/pt-br/fluttercountertutorial.md]:
```suggestion
?> **Nota**: `BlocBuilder` usa um parâmetro opcional `bloc`, mas podemos especificar o tipo do blocc e o tipo do estado, e o `BlocBuilder` encontrará o bloc automaticamente, assim não precisamos explicitar use `BlocProvider.of<CounterBloc>(context)`.
```

File: docs/pt-br/fluttercountertutorial.md [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
+```
+
+?> **Nota**: Podemos acessar a instância do `CounterBloc` usando o `BlocProvider.of <CounterBloc> (context)` porque envolvemos nossa `CounterPage` em um `BlocProvider`.
```
Comment (zepfietje) [inline docs/pt-br/fluttercountertutorial.md]:
```suggestion
?> **Nota**: Podemos acessar a instância do `CounterBloc` usando o `BlocProvider.of<CounterBloc>(context)` porque envolvemos nossa `CounterPage` em um `BlocProvider`.
```

**Comment by zepfietje** [comment]
@rodrigobastosv could you update all links like in https://github.com/felangel/bloc/pull/830?

File: docs/pt-br/fluttertimertutorial.md [hunk 12]
```diff
@@ -0,0 +1,1031 @@
...
+  }
+
+  Stream<TimerState> _mapPauseToState(Pause pause) async* {
+    if (state is Running) {
+      _tickerSubscription?.pause();
+      yield Paused(state.duration);
+    }
+  }
+
+  Stream<TimerState> _mapResumeToState(Resume pause) async* {
```
Comment (zepfietje) [inline docs/pt-br/fluttertimertutorial.md]: #832 

```suggestion
  Stream<TimerState> _mapResumeToState(Resume resume) async* {
```

File: docs/pt-br/fluttertimertutorial.md [hunk 13]
```diff
@@ -0,0 +1,1031 @@
...
+
+  Stream<TimerState> _mapPauseToState(Pause pause) async* {
+    final state = currentState;
+    if (state is Running) {
+      _tickerSubscription?.pause();
+      yield Paused(state.duration);
+    }
+  }
+
+  Stream<TimerState> _mapResumeToState(Resume pause) async* {
```
Comment (zepfietje) [inline docs/pt-br/fluttertimertutorial.md]: #832 

```suggestion
  Stream<TimerState> _mapResumeToState(Resume resume) async* {
```

File: docs/pt-br/fluttercountertutorial.md [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+description: A new Flutter project.
+version: 1.0.0+1
+
+environment:
+  sdk: ">=2.0.0 <3.0.0"
+
+dependencies:
+  flutter:
+    sdk: flutter
+  flutter_bloc: ^3.1.0
```
Comment (zepfietje) [inline docs/pt-br/fluttercountertutorial.md]:
```suggestion
  flutter_bloc: ^3.2.0
```

File: docs/pt-br/flutterfirebaselogintutorial.md [hunk 15]
```diff
@@ -0,0 +1,1986 @@
...
+environment:
+  sdk: ">=2.6.0 <3.0.0"
+
+dependencies:
+  flutter:
+    sdk: flutter
+  firebase_core: ^0.4.0+8
+  google_sign_in: ^4.0.0
+  firebase_auth: ^0.15.0+1
+  flutter_bloc: ^3.1.0
```
Comment (zepfietje) [inline docs/pt-br/flutterfirebaselogintutorial.md]:
```suggestion
  flutter_bloc: ^3.2.0
```

File: docs/pt-br/flutterfirestoretodostutorial.md [hunk 16]
```diff
@@ -0,0 +1,715 @@
...
+
+version: 1.0.0+1
+
+environment:
+  sdk: ">=2.6.0 <3.0.0"
+
+dependencies:
+  flutter:
+    sdk: flutter
+  flutter_bloc: ^3.1.0
```
Comment (zepfietje) [inline docs/pt-br/flutterfirestoretodostutorial.md]:
```suggestion
  flutter_bloc: ^3.2.0
```

File: docs/pt-br/flutterinfinitelisttutorial.md [hunk 17]
```diff
@@ -0,0 +1,607 @@
...
+
+version: 1.0.0+1
+
+environment:
+  sdk: ">=2.6.0 <3.0.0"
+
+dependencies:
+  flutter:
+    sdk: flutter
+  flutter_bloc: ^3.1.0
```
Comment (zepfietje) [inline docs/pt-br/flutterinfinitelisttutorial.md]:
```suggestion
  flutter_bloc: ^3.2.0
```

File: docs/pt-br/flutterlogintutorial.md [hunk 18]
```diff
@@ -0,0 +1,684 @@
...
+description: A new Flutter project.
+version: 1.0.0+1
+
+environment:
+  sdk: ">=2.6.0 <3.0.0"
+
+dependencies:
+  flutter:
+    sdk: flutter
+  flutter_bloc: ^3.1.0
```
Comment (zepfietje) [inline docs/pt-br/flutterlogintutorial.md]:
```suggestion
  flutter_bloc: ^3.2.0
```

File: docs/pt-br/fluttertimertutorial.md [hunk 19]
```diff
@@ -0,0 +1,1031 @@
...
+
+version: 1.0.0+1
+
+environment:
+  sdk: ">=2.6.0 <3.0.0"
+
+dependencies:
+  flutter:
+    sdk: flutter
+  flutter_bloc: ^3.1.0
```
Comment (zepfietje) [inline docs/pt-br/fluttertimertutorial.md]:
```suggestion
  flutter_bloc: ^3.2.0
```

File: docs/pt-br/fluttertodostutorial.md [hunk 20]
```diff
@@ -0,0 +1,1959 @@
...
+name: flutter_todos
+description: A new Flutter project.
+
+environment:
+  sdk: ">=2.6.0 <3.0.0"
+
+dependencies:
+  meta: ^1.1.6
+  equatable: ^1.0.0
+  flutter_bloc: ^3.1.0
```
Comment (zepfietje) [inline docs/pt-br/fluttertodostutorial.md]:
```suggestion
  flutter_bloc: ^3.2.0
```

File: docs/pt-br/flutterweathertutorial.md [hunk 21]
```diff
@@ -0,0 +1,2064 @@
...
+
+version: 1.0.0+1
+
+environment:
+  sdk: ">=2.6.0 <3.0.0"
+
+dependencies:
+  flutter:
+    sdk: flutter
+  flutter_bloc: ^3.1.0
```
Comment (zepfietje) [inline docs/pt-br/flutterweathertutorial.md]:
```suggestion
  flutter_bloc: ^3.2.0
```

File: docs/pt-br/flutterangulargithubsearch.md [hunk 22]
```diff
@@ -0,0 +1,1134 @@
...
+
+version: 1.0.0+1
+
+environment:
+  sdk: ">=2.6.0 <3.0.0"
+
+dependencies:
+  flutter:
+    sdk: flutter
+  flutter_bloc: ^3.1.0
```
Comment (zepfietje) [inline docs/pt-br/flutterangulargithubsearch.md]:
```suggestion
  flutter_bloc: ^3.2.0
```

**Comment by rodrigobastosv** [comment]
> @rodrigobastosv could you update all links like in #830?

Sure!

**Comment by rodrigobastosv** [comment]
> Will take a look tomorrow and merge  thanks so much for your hard work!

Thanks man. Will just do the @zepfietje sugestions!

File: docs/pt-br/flutterweathertutorial.md [hunk 23]
```diff
@@ -1315,7 +1315,7 @@ return RefreshIndicator(
 );
 ```
 
-Para usar o `RefreshIndicator`, tivemos que criar um [`Completer`](https://api.dartlang.org/stable/2.1.0/dart-async/Completer-class.html) o que nos permite produzir um "Future" que possamos concluir posteriormente.
+Para usar o `RefreshIndicator`, tivemos que criar um [`Completer`](https://api.dart.dev/stable/2.1.0/dart-async/Completer-class.html) o que nos permite produzir um "Future" que possamos concluir posteriormente.
```
Comment (zepfietje) [inline docs/pt-br/flutterweathertutorial.md]:
```suggestion
Para usar o `RefreshIndicator`, tivemos que criar um [`Completer`](https://api.dart.dev/stable/dart-async/Completer-class.html) o que nos permite produzir um "Future" que possamos concluir posteriormente.
```

File: docs/pt-br/gettingstarted.md [hunk 24]
```diff
@@ -19,15 +19,15 @@ dependencies:
   bloc: ^3.0.0
 ```
 
-Para uma aplicação [Flutter](https://flutter.io), também é necessário adicionar o package flutter_bloc como uma dependência no `pubspec.yaml`.
+Para uma aplicação [Flutter](https://flutter.dev), também é necessário adicionar o package flutter_bloc como uma dependência no `pubspec.yaml`.
```
Comment (zepfietje) [inline docs/pt-br/gettingstarted.md]:
```suggestion
Para uma aplicação [Flutter](https://flutter.dev/), também é necessário adicionar o package flutter_bloc como uma dependência no `pubspec.yaml`.
```

File: docs/pt-br/gettingstarted.md [hunk 25]
```diff
@@ -19,15 +19,15 @@ dependencies:
...
 
 ```yaml
 dependencies:
   bloc: ^3.0.0
   flutter_bloc: ^3.1.0
 ```
 
-Para uma aplicação [AngularDart](https://webdev.dartlang.org/angular), também é necessário adicionar o package angular_bloc como uma dependência no `pubspec.yaml`.
+Para uma aplicação [AngularDart](https://angulardart.dev), também é necessário adicionar o package angular_bloc como uma dependência no `pubspec.yaml`.
```
Comment (zepfietje) [inline docs/pt-br/gettingstarted.md]:
```suggestion
Para uma aplicação [AngularDart](https://angulardart.dev/), também é necessário adicionar o package angular_bloc como uma dependência no `pubspec.yaml`.
```

Comment (felangel) [reply docs/pt-br/gettingstarted.md]: why is the trailing slash necessary?

File: docs/pt-br/gettingstarted.md (hunk 24 above)
Comment (felangel) [reply docs/pt-br/gettingstarted.md]: why is the trailing slash necessary?

Comment (zepfietje) [reply docs/pt-br/gettingstarted.md]: It is not strictly necessary, but generally a trailing slash is used for the root path. Besides, it is nice to consistently stick to either having or not having a trailing slash (copying the URL will give you different results per browser).

File: docs/pt-br/gettingstarted.md (hunk 25 above)
Comment (zepfietje) [reply docs/pt-br/gettingstarted.md]: https://github.com/felangel/bloc/pull/813#discussion_r374004174

**Comment by felangel** [comment]
@rodrigobastosv are you able to make the minor changes requested by @zepfietje?

**Comment by rodrigobastosv** [comment]
> @rodrigobastosv are you able to make the minor changes requested by @zepfietje?

Done @felangel :)

**Pull request merged**